ERROR: Building tests failed

Building tests fails:

```
$ make
==> hamcrest (clean)
==> hamcrest (compile)
Compiled src/hamcrest.erl
Compiled src/hamcrest_term.erl
Compiled src/hamcrest_matchers.erl
==> hamcrest (post_compile)
==> hamcrest (xref)
==> proper (get-deps)
==> hamcrest (get-deps)
==> proper (compile)
make[1]: Wejście do katalogu '/home/aidecoe/Downloads/ejabberd-16.04/deps/hamcrest/deps/proper'
make[1]: 'include/compile_flags.hrl' jest aktualne.
make[1]: Opuszczenie katalogu '/home/aidecoe/Downloads/ejabberd-16.04/deps/hamcrest/deps/proper'
==> hamcrest (compile)
==> hamcrest (ct)
Showing log
--- Test run on 2016/05/28 16:23:48 ---
Converting "/home/aidecoe/Downloads/ejabberd-16.04/deps/hamcrest/." to "/home/aidecoe/Downloads/ejabberd-16.04/deps/hamcrest" and re-inserting with add_patha/1


Common Test v1.12 starting (cwd is /home/aidecoe/Downloads/ejabberd-16.04/deps/hamcrest)


Common Test: Running make in test directories...
Including the following directories:
"/home/aidecoe/Downloads/ejabberd-16.04/deps/hamcrest/include"
Recompile: hamcrest_matchers_SUITE
/home/aidecoe/Downloads/ejabberd-16.04/deps/hamcrest/include/hamcrest_internal.hrl:33: type set() undefined
/home/aidecoe/Downloads/ejabberd-16.04/deps/hamcrest/include/hamcrest_internal.hrl:34: type gb_set() undefined
hamcrest_matchers_SUITE.erl:72: Warning: variable 'Y' is unused
hamcrest_matchers_SUITE.erl:243: Warning: variable 'LStr' is unused
Recompile: hamcrest_SUITE
/home/aidecoe/Downloads/ejabberd-16.04/deps/hamcrest/include/hamcrest_internal.hrl:33: type set() undefined
/home/aidecoe/Downloads/ejabberd-16.04/deps/hamcrest/include/hamcrest_internal.hrl:34: type gb_set() undefined
{error,make_failed}


CWD set to: "/home/aidecoe/Downloads/ejabberd-16.04/deps/hamcrest/logs/ct_run.test@freja.2016-05-28_16.23.48"

TEST INFO: 1 test(s), 0 case(s) in 0 suite(s)

Testing deps.hamcrest: Starting test, 0 test cases

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
Error detected: 'hamcrest_SUITE can not be compiled or loaded'
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

Testing deps.hamcrest: *** FAILED {ct_framework,error_in_suite} ***

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
Error detected: 'hamcrest_matchers_SUITE can not be compiled or loaded'
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

Testing deps.hamcrest: *** FAILED {ct_framework,error_in_suite} ***
Testing deps.hamcrest: TEST COMPLETE, 0 ok, 0 failed of 0 test cases

Updating /home/aidecoe/Downloads/ejabberd-16.04/deps/hamcrest/logs/index.html... done
Updating /home/aidecoe/Downloads/ejabberd-16.04/deps/hamcrest/logs/all_runs.html... done

ERROR: Building tests failed
ERROR: ct failed while processing /home/aidecoe/Downloads/ejabberd-16.04/deps/hamcrest: rebar_abort
make: *** [Makefile:14: test] Błąd 1
```

Erlang version: 18.3

--- a/.gitignore
+++ b/.gitignore
@@ -4,6 +4,7 @@
 *.o
 *.so
 *.swp
+.rebar/
 build/*
 ^build$
 Emakefile
@@ -15,3 +16,10 @@ ebin/*.beam
 logs/*
 include/hamcrest.hrl
 qc.hrl
+deps/
+ct_*/
+all_runs.html
+ct_*
+jquery-latest.js
+jquery.tablesorter.min.js
+variables-ct@localhost

--- a/test.config
+++ b/test.config
@@ -1,9 +1,19 @@
-
 {cover_enabled, true}.
+
 {deps, [
-    %% TODO: bind to a specific commit or tag instead of 'master'
-    {proper, ".*", {git, "http://github.com/manopapad/proper.git", "v1.1"}}
+    {proper, "1.*", {git, "http://github.com/manopapad/proper.git", "v1.2"}}
 ]}.
+
+{erl_opts, [
+    debug_info,
+    fail_on_warning,
+    {src_dirs, ["test"]},
+    {platform_define, "^[0-9]+", namespaced_types}
+    ]}.
+
+{validate_app_modules, false}.
+
+{ct_extra_params, "-no_auto_compile"}.
 
 {plugin_dir, "priv/build/plugins"}.
 {plugins, [eqc_resolver]}.

--- a/test/hamcrest_matchers_SUITE.erl
+++ b/test/hamcrest_matchers_SUITE.erl
@@ -69,7 +69,7 @@ is_not_evaluates_to_logical_negation_of_underlying_matcher(_) ->
     ?EQC(P).
 
 is_not_provides_convenient_shortcut_for_not_equal_to(_) ->
-    P = ?FORALL({X, Y}, {binary(), binary()},
+    P = ?FORALL({X, _Y}, {binary(), binary()},
             begin
                 #'hamcrest.matchspec'{matcher=F1} = equal_to(X),
                 #'hamcrest.matchspec'{matcher=F2} = is_not(X),
@@ -240,7 +240,7 @@ ends_with_should_only_match_last_portion_of_string(_) ->
         ?IMPLIES(length(Xs) > 0,
         begin
           Y = round(length(Xs) / 2),
-          LStr = string:left(Xs, Y),
+          _LStr = string:left(Xs, Y),
           RStr = string:right(Xs, Y),
           case (assert_that(Xs, ends_with(RStr))) of
             true -> true;

